src/main/kotlin/org/rust/lang/core/completion/RsLifetimeCompletionProvider.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.completion
+
+import com.intellij.codeInsight.completion.CompletionParameters
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.patterns.ElementPattern
+import com.intellij.psi.PsiElement
+import com.intellij.util.ProcessingContext
+import org.rust.ide.refactoring.RsNamesValidator
+import org.rust.lang.core.RsPsiPattern
+
+object RsLifetimeCompletionProvider : RsCompletionProvider() {
```

Comment:
nit: Maybe rename somehow not to confuse readers. Currently, name looks like exactly this provider is responsible for completion of all lifetimes